helm-charts/splunk-otel-collector/templates/config/_common.tpl

```diff
@@ -80,8 +80,91 @@ resourcedetection:
     # The `system` detector goes last so it can't preclude cloud detectors from setting host/os info.
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor#ordering
     - system
+  {{- if hasPrefix "gke" (include "splunk-otel-collector.distribution" .) }}
+  gcp:
+    resource_attributes:
+      k8s.cluster.name:
+        enabled: true
+  {{- else if hasPrefix "eks" (include "splunk-otel-collector.distribution" .) }}
+  eks:
+    resource_attributes:
+      k8s.cluster.name:
+        enabled: true
+  {{- end }}
   override: true
+  {{/* Testing determined that the EKS cluster name detection required a slightly longer default timeout. */}}
+  {{- if hasPrefix "eks" (include "splunk-otel-collector.distribution" .) }}
+  timeout: 15s
+  {{- else -}}
   timeout: 10s
+  {{- end }}
```

Comment:
I think that's ok to increase timeout from 10s to 15s unconditionally. I won't affect the start time anyway